src/builtin.js

```diff
@@ -118,7 +133,10 @@ Sk.builtin.asnum$nofloat = function (a) {
     if (typeof a === "number") {
         a = a.toString();
     }
-    if (a.constructor === Sk.builtin.nmber) {
+    if (a.constructor === Sk.builtin.int_) {
+        a = a.v.toString();
+    }
+    if (a.constructor === Sk.builtin.float_) {
```

Comment:
nitpcik: can be written as `a.constructor === Sk.builtin.int_ || a.constructor === Sk.builtin.float_`